include/nrf_errno.h

```diff
@@ -0,0 +1,107 @@
+/*
```

Comment:
I am concern about the file name.

So, every module from nrfxlib, that needs errno, will include this file. If any c file will include this file and system's errno.h directly or indirectly, it will have two definition. This is ok for Zephyr, because we aligned all the numbers with it.

Problem is that nrfxlib is not only for Zephyr. I don't know about other modules that uses errno, but nRF RPC is system-independent and will allow calling remote procedures even from Linux (when serial backend will be finished). Linux errno.h is different than this, so nRF RPC will not compile on Linux if we switch to this solution.

My suggestion is to use name `errno.h`. Then, the build system must ensure that errno.h from system is used if available, and this file if not. I don't know how about binary modules. Does anyone allows to use them outside Zephyr, where system's `errno.h` may be incompatible? If yes, maybe we should consider some different approach?

Reply:
Yes, there are binary libraries which are not necessarily tied to Zephyr. When using those, the errno definitions must stay the same as those that the library was compiled with. What happens right now is that errnos are glued for those libraries, and I don't see a way around that.

I updated the PR to prefix all the errnos with the `NRF_` prefix.
I think that source libraries or libraries that are Zephyr-only may include just `errno.h` and be happy.
For those _binary_ libraries that are OS-independent, we'll have to glue so that the application can compile with whatever errno.h definitions that they have.

Reply:
Yes, from nRF RPC perspective, this is acceptable.

Reply:
We should talk with verification team to consider some automated tests that checks if all defined symbols are aligned with Zephyr. Now, we are aware, but in a few years maybe someone will want to change this file or Zephyr's and make them incompatible.

Different approach for testing would be put `BUILD_ASSERT`s in the file. I don't remember if there is a definition that allows us detect if we are compiling in Zephyr, but let's assume for this example that it is `IS_ZEPHYR`.

```c
#ifdef IS_ZEPHYR
#include <kernel.h>
#include <errno.h>
BUILD_ASSERT(NRF_EPERM == EPERM, "Invalid NRF_EPERM value");
BUILD_ASSERT(NRF_ENOENT == ENOENT, "Invalid NRF_ENOENT value");
...
```